src/lib.rs

```diff
@@ -483,49 +485,37 @@ impl Unicorn {
         }
     }
 
-    /// Write an unsigned value register.
-    ///
-    /// Note : The register is defined as an i32 to be able to support the
-    /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
-    /// You need to cast the register with `as i32`.
-    pub fn reg_write(&self, regid: i32, value: u64) -> Result<(), Error> {
-        let p_value: *const u64 = &value;
+    fn reg_write_generic<T: Sized>(&self, regid: i32, value: T) -> Result<()> {
+        let p_value: *const T = &value;
         let err = unsafe { uc_reg_write(self.handle, regid, p_value as *const libc::c_void) };
         if err == Error::OK {
             Ok(())
         } else {
             Err(err)
         }
     }
-    /// Write a signed 32-bit value to a register.
+
+    /// Write an unsigned value register.
     ///
     /// Note : The register is defined as an i32 to be able to support the
     /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
     /// You need to cast the register with `as i32`.
-    pub fn reg_write_i32(&self, regid: i32, value: i32) -> Result<(), Error> {
-        let p_value: *const i32 = &value;
-        let err = unsafe {
-            uc_reg_write(
-                self.handle,
-                regid as libc::c_int,
-                p_value as *const libc::c_void,
-            )
-        };
-        if err == Error::OK {
-            Ok(())
-        } else {
-            Err(err)
-        }
+    pub fn reg_write(&self, regid: i32, value: u64) -> Result<()> {
+        Self::reg_write_generic::<_>(&self, regid, value)
     }
 
-    /// Read an unsigned value from a register.
+    /// Write a signed 32-bit value to a register.
     ///
     /// Note : The register is defined as an i32 to be able to support the
     /// different register types (`RegisterX86`, `RegisterARM`, `RegisterMIPS` etc.).
     /// You need to cast the register with `as i32`.
-    pub fn reg_read(&self, regid: i32) -> Result<u64, Error> {
-        let mut value: u64 = 0;
-        let p_value: *mut u64 = &mut value;
+    pub fn reg_write_i32(&self, regid: i32, value: i32) -> Result<()> {
+        Self::reg_write_generic::<_>(&self, regid, value)
+    }
+
+    fn reg_read_generic<T: Sized + Default>(&self, regid: i32) -> Result<T> {
+        let mut value: T = Default::default();
```

Comment:
Use a `union` here and scrub `Default` bound.

Reply:
@strake thanks for the comment and sorry for my stupidity: I still don't understand how to use `union` to scrub `Default`

Reply:
See my [other comment](https://github.com/ekse/unicorn-rs/pull/45#issuecomment-456180217).